main.tf

```
@@ -361,3 +387,76 @@ resource "aws_s3_access_point" "ap" {
 
   depends_on = [var.module_depends_on]
 }
+
+# ---------------------------------------------------------------------------------------------------------------------
+# Replication Role and policy
+# ---------------------------------------------------------------------------------------------------------------------
+resource "aws_iam_role" "replication" {
+  count = local.replication_enabled ? 1 : 0
+  name  = "${var.bucket}-replication-role"
```

this should be given by the user.

```suggestion
  name       = try(var.replication.role_name, "${var.bucket}-replication-role")
```